scripts/tools/zap/zap_execution.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import subprocess
+import sys
+from typing import Tuple
+
+# The version MUST be of the form `YYYY.MM.YY'
+# Since this is ordered as such, alphabetical sorting will be used to check
+# validity
+#
+# This version MUST be changed together with `${ZAP_VERSION}` in
+#    - integrations/docker/images/chip-build/Dockerfile
+#
+# Potentially also change:
+#    - integrations/docker/images/chip-cert-bins/Dockerfile (in case new cert
+#      binaries also require a zap update)
+MIN_ZAP_VERSION = '2021.1.9'
+
+
+class ZapTool:
+    def __init__(self):
+        # Accepted environment variables, in order:
+        #
+        # ZAP_DEVELOPMENT_PATH - the path to a zap development environment. This is
+        #                        a zap checkout, used for local development
+        # ZAP_INSTALL_PATH     - the path where zap-cli exists. This is if zap-cli
+        #                        is NOT in the current path
+        if 'ZAP_DEVELOPMENT_PATH' in os.environ:
+            self.zap_start = ['node', 'src-script/zap-start.js']
+            self.working_directory = os.environ['ZAP_DEVELOPMENT_PATH']
+            # Make sure we don't try to munge the package.json in the ZAP repo.
+            os.environ['ZAP_SKIP_REAL_VERSION'] = '1'
```

Comment:
This is going to make the version checks fail, as observed when I tried this PR with ZAP_DEVELOPMENT_PATH....